### PR TITLE
BundleProbeTest: Enable disabled tests

### LINKED ISSUE #35755: Installer test BundleProbe.Bundle_Probe_Passed_For_Single_File_App failures on Linux_musl x64
### Builds
|Build|Pull Request | Test Failure Count|
| --- | --- | --- |
|[#626420](https://dev.azure.com/dnceng/public/_build/results?buildId=626420)|Rolling|1|
|[#626513](https://dev.azure.com/dnceng/public/_build/results?buildId=626513)|Rolling|1|
|[#626634](https://dev.azure.com/dnceng/public/_build/results?buildId=626634)|Rolling|1|
|[#626761](https://dev.azure.com/dnceng/public/_build/results?buildId=626761)|Rolling|1|
|[#626950](https://dev.azure.com/dnceng/public/_build/results?buildId=626950)|Rolling|1|
|[#627020](https://dev.azure.com/dnceng/public/_build/results?buildId=627020)|Rolling|1|
|[#627303](https://dev.azure.com/dnceng/public/_build/results?buildId=627303)|Rolling|1|
|[#627435](https://dev.azure.com/dnceng/public/_build/results?buildId=627435)|Rolling|1|
|[#627770](https://dev.azure.com/dnceng/public/_build/results?buildId=627770)|Rolling|1|
|[#627900](https://dev.azure.com/dnceng/public/_build/results?buildId=627900)|Rolling|1|
|[#627940](https://dev.azure.com/dnceng/public/_build/results?buildId=627940)|Rolling|1|
|[#627985](https://dev.azure.com/dnceng/public/_build/results?buildId=627985)|Rolling|1|
|[#627992](https://dev.azure.com/dnceng/public/_build/results?buildId=627992)|Rolling|1|
|[#628150](https://dev.azure.com/dnceng/public/_build/results?buildId=628150)|Rolling|1|
|[#628240](https://dev.azure.com/dnceng/public/_build/results?buildId=628240)|Rolling|1|
### Configurations
- Installer-coreclr-Linux_musl_x64-Release

```
Expected command to pass but it did not."\nFile Name: /root/runtime/artifacts/tests/Release/ahb/1/BundleProbeTester/bundle/BundleProbeTester\nArguments: SingleFile\nExit Code: 253\nStdOut:\nUnexpected file hostpolicy.dll found in bundle.\n\nStdErr:\n\n"

   at FluentAssertions.Execution.XUnit2TestFramework.Throw(String message)
   at FluentAssertions.Execution.TestFrameworkProvider.Throw(String message)
   at FluentAssertions.Execution.DefaultAssertionStrategy.HandleFailure(String message)
   at FluentAssertions.Execution.AssertionScope.FailWith(String message, Object[] args)
   at Microsoft.DotNet.CoreSetup.Test.CommandResultAssertions.Pass() in /_/src/installer/test/TestUtils/Assertions/CommandResultAssertions.cs:line 31
   at AppHost.Bundle.Tests.BundleProbe.Bundle_Probe_Passed_For_Single_File_App() in /_/src/installer/test/Microsoft.NET.HostModel.Tests/AppHost.Bundle.Tests/BundleProbe.cs:line 46
```

https://dev.azure.com/dnceng/public/_build/results?buildId=628240&view=ms.vss-test-web.build-test-results-tab&runId=19618302&resultId=100000&paneView=debug

Docker image: mcr.microsoft.com/dotnet-buildtools/prereqs:alpine-3.9-WithNode-0fc54a3-20190918214015

@swaroop-sridhar this is blocking clean rolling builds, can you please take a look?  

cc: @dotnet/runtime-infrastructure

### PR DESCRIPTION
**Pull request by swaroop-sridhar**
A few of the bundle probe tests were failing on Linux-musl-x64-release.
This change adjusts the test to circumvent the failure.

Fixes #35755

### CODE REVIEW

File: src/installer/test/Assets/TestProjects/BundleProbeTester/Program.cs [hunk 1]
```diff
@@ -9,13 +9,17 @@ namespace BundleProbeTester
 {
     public static class Program
     {
+        // The return type on BundleProbeDelegate is byte instead of bool because
+        // using non-blitable bool type caused a failure (incorrect value) on linux-musl-x64.
+        // The bundle-probe callback is only called from native code in the product
+        // Therefore the type on this test is adjusted to circumvent the failure.
         [UnmanagedFunctionPointer(CallingConvention.StdCall)]
-        public delegate bool BundleProbeDelegate([MarshalAs(UnmanagedType.LPWStr)] string path, IntPtr size, IntPtr offset);
+        public delegate byte BundleProbeDelegate([MarshalAs(UnmanagedType.LPWStr)] string path, IntPtr size, IntPtr offset);
```
Comment (VSadov) [inline src/installer/test/Assets/TestProjects/BundleProbeTester/Program.cs:17]: Is this ok to be `LPWstr` or should become `IntPtr`? 
If the goal is to inline or avoid a stub, can that happen if the call needs to pin the string?

**Comment by VSadov** [comment]
Will existing scenarios use this code path?  (without singlefilehost).
I mean - the tests are passing. Does it mean we are all ok here?

**Review by AaronRobinsonMSFT — APPROVED** [review]

**Review by VSadov — APPROVED** [review]

File: src/installer/test/Assets/TestProjects/BundleProbeTester/Program.cs (hunk 1 above)
Comment (swaroop-sridhar) [reply src/installer/test/Assets/TestProjects/BundleProbeTester/Program.cs:17]: Yes the `LPWstr` is OK.
I spoke with @jkoritzinsky  about this today, looks like the problem was wrt managed `bool` vs native `bool` values in the return type.  Please see: https://github.com/dotnet/runtime/issues/36372.

**Comment by swaroop-sridhar** [comment]
> Will existing scenarios use this code path? (without singlefilehost).
> I mean - the tests are passing. Does it mean we are all ok here?

In the non-single-file case, we simply check that the `BUNDLE_PROBE` property is [not passed](https://github.com/dotnet/runtime/blob/master/src/installer/test/Assets/TestProjects/BundleProbeTester/Program.cs#L51-L61) to the runtime.

**Pull request merged**